bigquery/bigquery_datatransfer_quickstart/main.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2017 Google LLC. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+// [START bigquery_datatransfer_quickstart]
+// Sample bigquery-quickstart creates a Google BigQuery dataset.
+package main
+
+import (
+	"fmt"
+	"log"
+
+	// Imports the BigQuery Data Transfer client package.
+	"cloud.google.com/go/bigquery/datatransfer/apiv1"
+	"golang.org/x/net/context"
+	"google.golang.org/api/iterator"
+	datatransferpb "google.golang.org/genproto/googleapis/cloud/bigquery/datatransfer/v1"
+)
+
+func main() {
+	ctx := context.Background()
+
+	// Sets your Google Cloud Platform project ID.
+	projectID := "YOUR_PROJECT_ID"
+
+	// Creates a client.
+	client, err := datatransfer.NewClient(ctx)
+	if err != nil {
+		log.Fatalf("Failed to create client: %v", err)
+	}
+
+	req := &datatransferpb.ListDataSourcesRequest{
+		Parent: fmt.Sprintf("projects/%s", projectID),
+	}
+	it := client.ListDataSources(ctx, req)
+	fmt.Println("Supported Data Sources:")
+	for {
+		resp, err := it.Next()
```

Comment:
I might rename "resp" to something that reflects the return type of `*datatransferpb.DataSource`. Maybe `ds`?